First pass at auth0 integration docs

Driptap opened the pull request:
I have a working Auth0 integration and wanted to share it, this is a first pass and explaining it in documentation I would have found useful when setting it up.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Documentation**
	- Enhanced content and structure of the Auth0 integration guide.
	- Added a new introductory section and detailed examples for better clarity.
	- Introduced sections on JWT authentication and user management.
	- Improved error handling for unauthenticated users.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Driptap commented on docs/guides/authentication/auth0.md:
````diff
@@ -4,6 +4,129 @@ sidebar_position: 6
+Here's an example using a JWT:
+
+```
+export const getPrismaJWT = async (req) => {
+  try {
+    const jwks = jose.createRemoteJWKSet(new URL(process.env.AUTH0_JWKS_URI));
+    const token = toString(req.headers.get('authorization')).replace('Bearer ', '');
+    const res = await jose.jwtVerify(token, jwks, {
+      issuer: `${process.env.AUTH0_ISSUER_BASE_URL}/`,
+      audience: process.env.AUTH0_TWIST_API_AUDIENCE,
+      algorithms: ['RS256'],
+    });
+  
+    const userId = res.payload.sub;
+    const user = {
+      id: userId,
+      specialKey: res.payload.metadata.specialKey
+    };
+  
+    return enhance(prisma, {user});
+  catch (err) {
+    // unauthenticated error
+  }  
+};
````
This is an example mr.Rabbit

ymc9 commented on docs/guides/authentication/auth0.md:
```diff
@@ -4,6 +4,129 @@ sidebar_position: 6
+
+## Adding Auth0 authentication
+
+You can authenticate an Auth0 user and extract information from the authentication and apply is to the Zenstack session.
```
"apply is to" should be "apply it to"?

ymc9 commented on docs/guides/authentication/auth0.md:
````diff
@@ -4,6 +4,129 @@ sidebar_position: 6
+
+Here's an example using a JWT:
+
+```
````
Marking the code block as TS will enable syntax highlighting:

\`\`\`ts
export const getPrismaJWT = ...
\`\`\`

ymc9 commented on docs/guides/authentication/auth0.md:
```diff
@@ -4,6 +4,129 @@ sidebar_position: 6
+
+When the client is created, the database is queried using the contents of the Auth0 token.
+
+In this case, the Auth tyoe is what provide authentication, not the User model, for example: 
```
"tyoe" => "type"?

ymc9 commented on docs/guides/authentication/auth0.md:
```diff
@@ -4,6 +4,129 @@ sidebar_position: 6
+    where: { id: session.user.sub },
+  }); 
+  
+  return {
```
I'm not following why this triggers the creating of a new user. Is the `getPrisma` function supposed to do it?

Driptap replied:
Ah no - i was trying to show how the non-existence of the database record could be used to trigger onboarding behaviour such as redirecting to a signup form. I'll try and make it clearer.

ymc9 commented on docs/guides/authentication/auth0.md:
````diff
@@ -4,6 +4,129 @@ sidebar_position: 6
+This would populate your `auth()` in the Zmodel with the object you've just created from auth0; enabling checks like:
+
+```
+@@allow('read, update, create', auth().id == this.id)
````
Likewise we can mark this code block with \`\`\`prisma \`\`\`

ymc9 reviewed (changes requested):
Hi @Driptap , many thanks for working on this guide! Auth0 integration has been in the backlog for quite a while and finally we'll have it!

I've added a couple of comments. Please help check them. Thanks!

Driptap commented:
Thanks for your review @ymc9 - i think i've addressed all your comments. It's a little tricky gauging how much detail to go into in the section "Working along side a user model". I've added some more details & examples to it to try to make it clearer.

ymc9 commented:
> Thanks for your review @ymc9 - i think i've addressed all your comments. It's a little tricky gauging how much detail to go into in the section "Working along side a user model". I've added some more details & examples to it to try to make it clearer.

Awesome, thank you! I've made some final touchup. Merging it now!
